# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [14]:
import pandas as pd
import numpy as np
from scipy import stats as st

In [4]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [8]:
print(data.columns)
print(data.shape)

Index([u'id', u'ad', u'education', u'ofjobs', u'yearsexp', u'honors',
       u'volunteer', u'military', u'empholes', u'occupspecific', u'occupbroad',
       u'workinschool', u'email', u'computerskills', u'specialskills',
       u'firstname', u'sex', u'race', u'h', u'l', u'call', u'city', u'kind',
       u'adid', u'fracblack', u'fracwhite', u'lmedhhinc', u'fracdropout',
       u'fraccolp', u'linc', u'col', u'expminreq', u'schoolreq', u'eoe',
       u'parent_sales', u'parent_emp', u'branch_sales', u'branch_emp', u'fed',
       u'fracblack_empzip', u'fracwhite_empzip', u'lmedhhinc_empzip',
       u'fracdropout_empzip', u'fraccolp_empzip', u'linc_empzip', u'manager',
       u'supervisor', u'secretary', u'offsupport', u'salesrep', u'retailsales',
       u'req', u'expreq', u'comreq', u'educreq', u'compreq', u'orgreq',
       u'manuf', u'transcom', u'bankreal', u'trade', u'busservice',
       u'othservice', u'missind', u'ownership'],
      dtype='object')
(4870, 65)


In [6]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

## Question #1:
## What test is appropriate for this problem? Does CLT apply?


#### - The data does provide over 30 samples, so a 2 sample z test is appropriate for this one. CLT does apply as the number of samples provided is over 10% of the population and the samples are independent

## Question #2:
## What are the null and alternate hypotheses?


#### - Null Hypothesis H0: Race has no significant impact on the rate of callbacks for resumes.
#### - Alternate Hypothesis H1: Race has a significant impact on the rate of callbacks for resumes.

## Question #3:
### Compute margin of error, confidence interval, and p-value.


In [18]:
black = data.loc[data['race']=='b']
white = data.loc[data['race']=='w']

b_prob = black['call'].sum()/black.shape[0]
print("Proportion of w race getting a callback:")
print(b_prob)

w_prob = white['call'].sum()/white.shape[0]
print("Proportion of b race getting a callback:") 
print(w_prob)

Proportion of w race getting a callback:
0.064476386037
Proportion of b race getting a callback:
0.0965092402464


In [15]:
std = np.sqrt(b_prob*(1-b_prob)/black.shape[0]+w_prob*(1-w_prob)/white.shape[0])
margin_err = std*2
conf_int = [-margin_err, margin_err]
prob_diff = b_prob-w_prob

p_val = st.norm.cdf(prob_diff/std)
print("margin of error: {}".format(margin_err))
print("confidence interval: {}".format(conf_int))
print("p-value: {}".format(p_val))

margin of error: 0.0155667411734
confidence interval: [-0.015566741173353509, 0.015566741173353509]
p-value: 1.93128260376e-05


## Question #4:
### Write a story describing the statistical significance in the context or the original problem.


- There is a significant difference in means from call backs of white sounding names compared to black sounding names.
This difference can be seen in the samples where 9.65% of white sounding names get called back compared to only 6.45% of black sounding names.

## Question #5:
### Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

 - The results of this study indicate that, all other things being equal, race is still an important factor in the American labor market. An African American applicant' s race certainly has negative effects on his employment prospects on average. Resumes with white-sounding names received 50 percent more callbacks than those with black names.